### ISSUE TITLE
Adding index does not speed up SQL execution

### ISSUE THREAD

**Issue by bluestreak01** [questdb/questdb#1939]
### Describe the bug

My symbol lookup filter was painfully slow. I decided to add index on said symbol. Query remained slow until instance restart. After restart query is much faster.

### Environment

```markdown
- **QuestDB version**: 6.2.1
- **OS**:
- **Browser**:
```

**Comment by puzpuzpuz** [comment]
Loosely related with #1705. That's because add index statement has to invalidate all cached queries (ideally only for the affected table, but invalidating all of them may be acceptable).

**Comment by puzpuzpuz** [comment]
> Query remained slow until instance restart. After restart query is much faster.

Are you sure that it's the case? I conducted some experiments and it seems to be that all queries that use the modified table should get recompiled because of a `ReaderOutOfDateException` exception thrown when obtaining a `TableReader`. The exception would be thrown since the table version has changed when compared to what's in the pooled reader. WDYT?

https://github.com/questdb/questdb/blob/919ebf629120f92f09907a35d024a1d3e1a3979f/core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java#L2314-L2319

https://github.com/questdb/questdb/blob/6eca37b8f27bb1d51efc6cd8e68199ef2bd57b82/core/src/main/java/io/questdb/cutlass/http/processors/JsonQueryProcessor.java#L161-L164

https://github.com/questdb/questdb/blob/618fce56c396d54d070b91d1e6dabf2e1494cb17/core/src/main/java/io/questdb/cairo/CairoEngine.java#L283

**Comment by bluestreak01** [comment]
I’m not sure, I tried to capture users feedback in the description. Perhaps there are outliers from the rule? Does adding index always bump correct version?

**Comment by puzpuzpuz** [comment]
> Does adding index always bump correct version?

Yes, it does and the bump is non-conditional: https://github.com/questdb/questdb/blob/ed76490b452a8ae6e2b045fe68a5d812611fcf05/core/src/main/java/io/questdb/cairo/TableWriter.java#L571

Is it possible that one of factories (or cursors) caches the `TableReader` and doesn't call `ReaderPool#get()` on `getCursor()` call? Seems very unlikely to me and I failed to find any such (broken) factories in the code.
